presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,470 @@
+import static io.prestosql.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.decimalColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.realWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.timestampWriteFunctionUsingSqlTimestamp;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varbinaryWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
+import static io.prestosql.plugin.jdbc.TypeHandlingJdbcSessionProperties.getUnsupportedTypeHandling;
+import static io.prestosql.plugin.jdbc.UnsupportedTypeHandling.CONVERT_TO_VARCHAR;
+import static io.prestosql.plugin.jdbc.UnsupportedTypeHandling.IGNORE;
+import static io.prestosql.spi.StandardErrorCode.ALREADY_EXISTS;
+import static io.prestosql.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static io.prestosql.spi.type.VarbinaryType.VARBINARY;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.Varchars.isVarcharType;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.sql.ResultSetMetaData.columnNoNulls;
+import static java.util.Collections.nCopies;
+import static java.util.Locale.ENGLISH;
+import static java.util.stream.Collectors.joining;
+
+public class ClickHouseClient
+        extends BaseJdbcClient
+{
+    private static final Logger log = Logger.get(ClickHouseClient.class);
+    private final Type jsonType;
+    public static final String ESCAPE_CHARACTER = "\"";
+
+    @Inject
+    public ClickHouseClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, TypeManager typeManager)
+    {
+//        super(config, "`", connectionFactory);
+
+        super(
+                ESCAPE_CHARACTER,
+                connectionFactory,
+                ImmutableSet.of(),
+                config.isCaseInsensitiveNameMatching(),
+                config.getCaseInsensitiveNameMatchingCacheTtl());
+        this.jsonType = typeManager.getType(new TypeSignature(StandardTypes.JSON));
+    }
+
+    @Override
+    public void abortReadConnection(Connection connection)
+            throws SQLException
+    {
+        connection.abort(directExecutor());
+    }
+
+    @Override
+    public void finishInsertTable(JdbcIdentity identity, JdbcOutputTableHandle handle)
+    {
+        String temporaryTable = quoted(null, handle.getSchemaName(), handle.getTemporaryTableName());
```

Comment:
Instead of overriding finishInsert, can we override `quoted` to skip the catalogName ?